server/src/test/java/org/elasticsearch/common/regex/RegexTests.java

```diff
@@ -78,6 +78,14 @@ public void testDoubleWildcardMatch() {
         assertFalse(Regex.simpleMatch("fff******ddd", "fffabcdd"));
     }
 
+    public void testArbitraryWildcardMatch() {
```

Comment:
Nit: to cover all branches, consider adding a use case that explicitly use multiple wildcards at the end of the pattern (maybe to the test before this one), and a use case where you have multiple wildcards in multiple locations (like dd**dd**dd).

Reply:
Thanks for the suggestion. I've initially randomized the test which should cover `dd****`, `****dd` and `dd****dd` cases but not `dd**dd**dd`. I've changed the test to be more explicit in testing all cases in a single run instead of depending on randomization.